Index: linux-2.6.31.6/arch/mips/kernel/Makefile
===================================================================
--- linux-2.6.31.6.orig/arch/mips/kernel/Makefile	2009-11-14 15:01:31.000000000 +0100
+++ linux-2.6.31.6/arch/mips/kernel/Makefile	2009-11-14 17:19:38.000000000 +0100
@@ -4,6 +4,8 @@
 
 CPPFLAGS_vmlinux.lds := $(KBUILD_CFLAGS)
 
+CPPFLAGS_vmlinux.lds := $(KBUILD_CFLAGS) $(EXTRA_LDSFLAGS)
+
 extra-y		:= head.o init_task.o vmlinux.lds
 
 obj-y		+= cpu-probe.o branch.o entry.o genex.o irq.o process.o \
